boomerang/src/nicp_to_icp.rs

```diff
@@ -11,10 +11,11 @@ use icrc_ledger_types::icrc1::transfer::TransferArg;
 use icrc_ledger_types::icrc2::approve::ApproveArgs;
 
 pub async fn notify_nicp_deposit(target: Principal) -> Result<WithdrawalSuccess, BoomerangError> {
+    let boomerang_id = self_canister_id();
     let subaccount = derive_subaccount_unstaking(target);
 
     let boomerang_account = Account {
-        owner: self_canister_id(),
+        owner: boomerang_id,
```

Comment:
why do you want to add a line?

Reply:
It is out of consistency. I think it should be either:
```
let boomerang_id = self_canister_id();
    let subaccount = derive_subaccount_staking(target);

    let boomerang_account = Account {
        owner: boomerang_id,
        subaccount: Some(subaccount),
    };
``` 

or
```
    let boomerang_account = Account {
        owner: self_canister_id(),
        subaccount: Some(derive_subaccount_staking(target),
    };
```

But not the in between computation that was there before.